msbuild/Xamarin.Shared/Xamarin.Shared.targets

```diff
@@ -912,6 +912,124 @@ Copyright (C) 2018 Microsoft. All rights reserved.
 		</PrepareResourceRules>
 	</Target>
 
+	<!-- App extensions -->
+
+	<Target Name="_AssignAppExtensionConfiguration" Condition="'@(_AppExtensionReference)' != ''">
+		<!-- assign configs if building a solution file -->
+		<AssignProjectConfiguration
+			ProjectReferences = "@(_AppExtensionReference)"
+			SolutionConfigurationContents = "$(CurrentSolutionConfigurationContents)"
+			Condition="'$(CurrentSolutionConfigurationContents)' != ''">
+
+			<Output TaskParameter="AssignedProjects" ItemName="_AppExtensionReferenceWithConfiguration"/>
+		</AssignProjectConfiguration>
+
+		<!-- Else, just -->
+		<CreateItem Include="@(_AppExtensionReference)"
+					Condition="'$(CurrentSolutionConfigurationContents)' == ''">
+			<Output TaskParameter="Include" ItemName="_AppExtensionReferenceWithConfiguration"/>
+		</CreateItem>
+	</Target>
+
+	<Target Name="_SeparateAppExtensionReferences" BeforeTargets="AssignProjectConfiguration">
```

Comment:
I know it was like this in the old target files but shouldn't the BeforeTargets be `_AssignProjectConfiguration` instead of `AssignProjectConfiguration`?

Reply:
I think that refers to the `AssignProjectConfiguration` target shipped by the msbuild: https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-targets?view=vs-2019 (we don't have a `_AssignProjectConfiguration`, and I couldn't find it anywhere else either).